server/model/model.go

```diff
@@ -0,0 +1,182 @@
+func New(txn store.FsTxn, keys map[string]*v1manifest.KeyInfo) Model {
+	return &model{txn, keys}
+}
+
+func (m *model) UpdateComponentManifest(component string, manifest *ComponentManifest) error {
+	snap, err := m.ReadSnapshotManifest()
+	if err != nil {
+		return err
+	}
+	lastVersion := snap.Signed.Meta[fmt.Sprintf("/%s.json", component)].Version
+	if manifest.Signed.Version != lastVersion+1 {
+		log.Debugf("Component version not expected, expect %d, got %d", lastVersion+1, manifest.Signed.Version)
+		return ErrorConflict
+	}
+	if err := m.txn.WriteManifest(fmt.Sprintf("%d.%s.json", manifest.Signed.Version, component), manifest); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (m *model) UpdateRootManifest(manifest *RootManifest) error {
+	var last RootManifest
+	if err := m.txn.ReadManifest("root.json", &last); err != nil {
+		return err
+	}
+	if manifest.Signed.Version != last.Signed.Version+1 {
+		return ErrorConflict
+	}
+	if err := m.txn.WriteManifest("root.json", manifest); err != nil {
+		return err
+	}
+	return m.txn.WriteManifest(fmt.Sprintf("%d.root.json", manifest.Signed.Version), manifest)
+}
+
+func (m *model) UpdateIndexManifest(f func(*IndexManifest) *IndexManifest) error {
+	snap, err := m.ReadSnapshotManifest()
+	if err != nil {
+		return err
+	}
+	lastVersion := snap.Signed.Meta["/index.json"].Version
+
+	var last IndexManifest
+	if err := m.txn.ReadManifest(fmt.Sprintf("%d.index.json", lastVersion), &last); err != nil {
+		return err
+	}
+	manifest := f(&last)
+	manifest.Signed.Version = last.Signed.Version + 1
+	manifest.Signatures, err = sign(manifest.Signed, m.keys[v1manifest.ManifestTypeIndex])
+	if err != nil {
+		return err
+	}
+
+	if err := m.txn.WriteManifest("index.json", manifest); err != nil {
+		return err
+	}
+	return m.txn.WriteManifest(fmt.Sprintf("%d.index.json", manifest.Signed.Version), manifest)
+}
+
+func (m *model) UpdateSnapshotManifest(f func(*SnapshotManifest) *SnapshotManifest) error {
+	var last SnapshotManifest
+	err := m.txn.ReadManifest("snapshot.json", &last)
+	if err != nil {
+		return err
+	}
+	manifest := f(&last)
+	manifest.Signatures, err = sign(manifest.Signed, m.keys[v1manifest.ManifestTypeSnapshot])
+	if err != nil {
+		return err
+	}
+
+	return m.txn.WriteManifest("snapshot.json", manifest)
+}
+
+// ReadSnapshotManifest returns snapshot.json
+func (m *model) ReadSnapshotManifest() (*SnapshotManifest, error) {
+	var snap SnapshotManifest
+	if err := m.txn.ReadManifest("snapshot.json", &snap); err != nil {
+		return nil, err
+	}
+	return &snap, nil
+}
+
+func (m *model) UpdateTimestampManifest() error {
+	fi, err := m.txn.Stat("snapshot.json")
+	if err != nil {
+		return err
+	}
+	reader, err := m.txn.Read("snapshot.json")
+	if err != nil {
+		return err
+	}
+	sha256, err := utils.SHA256(reader)
+	if err != nil {
+		reader.Close()
+		return err
+	}
+	reader.Close()
+
+	var manifest TimestampManifest
+	err = m.txn.ReadManifest("timestamp.json", &manifest)
```

Comment:
There is a constant definition for string literal `timestamp.json` in the repository package.